lib/addins/Misc2000GarageDoorOpener.js

```diff
@@ -45,66 +61,130 @@ class Misc2000GarageDoorOpener extends HandlerPattern {
 
 		console.log('INFO: on KNX Value Change(' + field + ", old="+ oldValue + ", new="+ knxValue+ ")");
 
+	//	this.myAPI.getValue("CurrentDoorState") // Read last HK Status first
+
 		if (field==="KNX_Garagentor_Move") {
 			// KNX_Garagentor_Move ist bei meiner Installation Hoch/Stopp/Runter im Wechsel
 			if (knxValue===1) {
 				// Befehl 1 auf BUS erkannt
 				if (oldValue!==1){ // Nur wenn Wert sich geändert hat
 					console.log('INFO: KNX_Garagentor_Move erkannt');
-//					this.myAPI.setValue("LockTargetState", 1); // SECURED
 
-					if (this.myAPI.getValue("KNX_Garagentor_zu")===0){
-						console.log('INFO: TargetDoorState = 0 OPEN');
+					if (this.timer1 !== undefined){
+						clearTimeout(this.timer1);
+						this.timer1 = undefined;
+						console.log('INFO: Timer für KNX_Garagentor_auf zurückgesetzt');
+					}
+
+					if (this.myAPI.getValue("KNX_Garagentor_zu")===0){ // IF DOOR IS CLOSED
+					// also CurrentDoorState = 1
+						console.log('INFO: Aktueller Status CurrentDoorState = 0 CLOSED erkannt');
+						console.log('INFO: TargetDoorState = 0 OPEN setzen');
 						this.myAPI.setValue("TargetDoorState", 0); // OPEN
-						console.log('INFO: CurrentDoorState = 2 OPENING');
+						console.log('INFO: CurrentDoorState = 2 OPENING setzen');
 						this.myAPI.setValue("CurrentDoorState", 2); // OPENING
-					} // ("KNX_Garagentor_zu")==1
+						this.lastMove = "goingUp";
+						console.log('INFO: lastMove = ' + this.lastMove+ ' setzen');
+						if (this.simulateDoorOpenContact){
+							this.myAPI.knxWrite("KNX_Garagentor_auf", 1, "DPT1"); // Tor nicht auf
+							console.log('INFO: Timer für KNX_Garagentor_auf gestartet');
+							this.timer1 = setTimeout (function (field) {
+							// 18 Seconds Timer
+													// Timer abgelaufen
+													console.log('INFO: Timer abgelaufen, Tor oben angekommen wird simuliert');
+													console.log('INFO: KNX_Garagentor_auf = 0 setzen');
+													this.myAPI.knxWrite("KNX_Garagentor_auf", 0, "DPT1"); // Tor oben angekommen
+							}.bind(this),20000, field);
+						} // simulateDoorOpenContact
+					} // ("KNX_Garagentor_zu")===1
 
+					else if (this.myAPI.getValue("CurrentDoorState")===2){ // IF DOOR IS OPENING
+						console.log('INFO: Aktueller Status CurrentDoorState = 2 OPENING erkannt');
+						console.log('INFO: CurrentDoorState = 4 STOPPED setzen');
+						this.myAPI.setValue("CurrentDoorState", 4); // STOPPED
+					} // ("CurrentDoorState")===2 //OPENING
 
-					if (this.myAPI.getValue("CurrentDoorState")===2){ //OPENING
-						console.log('INFO: CurrentDoorState = 4 STOPPED');
+					else if (this.myAPI.getValue("CurrentDoorState")===3){ // IF DOOR IS CLOSING
+						console.log('INFO: Aktueller Status CurrentDoorState = 3 CLOSING erkannt');
+						console.log('INFO: CurrentDoorState = 4 STOPPED setzen');
 						this.myAPI.setValue("CurrentDoorState", 4); // STOPPED
```

Comment:
OK you are right this could make problems if JS done execute the if else right. I will change it tomorrow. But first I need to solve the problem with the .getValue() 
But as a also try to log out the value of CurrentDoorState and always got unknown it looks like I  tested with the faulty version the last 2 days...